src/metabase/util.clj

```diff
@@ -725,3 +728,25 @@
   (fn [& args]
     (apply xor (for [pred preds]
                  (apply pred args)))))
+
+(defn topological-sort
+  "Topologically sorts vertexs in graph g.
+   https://en.wikipedia.org/wiki/Topological_sorting"
```

Comment:
Can we get a little example usage? I'm having a hard time picturing how to use this without seeing it

Also, is this general enough to live in `metabase.util`? My thinking is stuff that is in `metabase.util` is stuff that could go in a Metabase version of Medley if we were to spin it off. There's definitely some stuff in there that doesn't fit that criteria (e.g. `u/host-up?`). I've been trying to work on moving those elsewhere to keep this namespace full of super-general things but it's an ongoing project

But it might make more sense to make a separate entities utils namespace for this sort of stuff 🤷‍♂

Reply:
The followup PRs use it in some more places. I'm not super set on it being in util but it's one of those things that crops up occasionally as it's also useful for cycle detection.